quickwit/quickwit-config/src/config.rs

```diff
@@ -880,4 +894,41 @@ mod tests {
             config_builder.build().await.unwrap_err();
         }
     }
+
+    #[tokio::test]
+    async fn test_quickwit_config_data_dir_accepts_both_file_uris_and_file_paths() {
+        {
+            let config_yaml = r#"
+                version: 0
+                data_dir: /opt/quickwit/data
+            "#;
+            let config_builder =
+                serde_yaml::from_str::<QuickwitConfigBuilder>(config_yaml).unwrap();
+            let config = config_builder.build().await.unwrap();
+            assert_eq!(config.data_dir_path, PathBuf::from("/opt/quickwit/data"));
+        }
+        {
+            let config_yaml = r#"
+                version: 0
+                data_dir: file:///opt/quickwit/data
+            "#;
+            let config_builder =
+                serde_yaml::from_str::<QuickwitConfigBuilder>(config_yaml).unwrap();
+            let config = config_builder.build().await.unwrap();
+            assert_eq!(config.data_dir_path, PathBuf::from("/opt/quickwit/data"));
+        }
+        {
+            let config_yaml = r#"
+                version: 0
+                data_dir: s3://indexes/foo
+            "#;
+            let config_builder =
+                serde_yaml::from_str::<QuickwitConfigBuilder>(config_yaml).unwrap();
+            let error = config_builder.build().await.unwrap_err();
```

Comment:
Testing error messages is brittle. Searching for a few important words in the error message with `.contains()` is more robust or simply use `unwrap_err` when that makes sense.